### PR TITLE
Action triggers for node client

### PR DESCRIPTION
**Pull request by javo**
Action triggers types:
- Exact time: Runs one time at specific date and time.
- Repeat time: Runs at specific time on determined days of the week until a day the users define.
- Events: Runs when a specific event is triggered inside the client, such as 'stopped_charging', 'geofencing_in', etc

### CODE REVIEW

File: lib/agent/actions/triggers/index.js [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+      var rule = new schedule.RecurrenceRule();
+      rule.second    = info.second || 0;
+      rule.minute    = info.minute || null;
+      rule.hour      = info.hour   || null;
+      rule.dayOfWeek = info.days_of_week || null;
+
+      opts = {rule: rule};
+
+      if (info.until) {
+        // Si ya pasó la fecha definida en until retornar false
```
Comment (patriciojofre) [inline lib/agent/actions/triggers/index.js]: spanish? :-p

Comment (javo) [reply lib/agent/actions/triggers/index.js]: uuuhh, se me pasó eso, era un recordatorio de un caso específico.

File: test/lib/agent/actions/fixtures/triggers_responses.js [hunk 2]
```diff
@@ -0,0 +1,245 @@
+var exact_triggers = [
+  {
+    "id":105,
+    "name":"trigger 105",
+    "events":[
+      {
+        "type": "exact_time",
+        "info": {
+          "date" : "20190725150406"   // CREAR FUNCION PARA GENERAR FECHAS SIEMPRE EN EL FUTURO
```
Comment (patriciojofre) [inline test/lib/agent/actions/fixtures/triggers_responses.js]: @javo ☝️

File: test/lib/agent/actions/fixtures/triggers_responses.js [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+   },
+   
+   {
+    "id":106,
+    "name":"trigger 106",
+    "events":[
+      {
+        "type": "exact_time",
+        "info": {
+          "date" : "2019-07-25T15:04:07Z-04:00"    // ACA TB
```
Comment (patriciojofre) [inline test/lib/agent/actions/fixtures/triggers_responses.js]: same here

File: test/lib/agent/actions/triggers.js [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+          storage.all('triggers', (err, obj) => {
+            Object.keys(obj).length.should.be.equal(0)
+            done();
+          })
+        })
+      })
+
+      describe('and has one or more trigger', () => {
+
+        /*
```
Comment (patriciojofre) [inline test/lib/agent/actions/triggers.js]: remove this comment ¿?

File: test/lib/agent/actions/triggers.js [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+            clock.tick(2000)
+            storage.all('triggers', (err, obj) => {
+              Object.keys(obj).length.should.be.equal(5)
+              done();
+            })
+          })
+        })
+
+        describe('and it has all type of triggers', () => {
+
```
Comment (patriciojofre) [inline test/lib/agent/actions/triggers.js]: TODO ¿?

File: lib/agent/actions/triggers/index.js (hunk 1 above)
Comment (javo) [reply lib/agent/actions/triggers/index.js]: fixed

File: test/lib/agent/actions/fixtures/triggers_responses.js (hunk 2 above)
Comment (javo) [reply test/lib/agent/actions/fixtures/triggers_responses.js]: also fixed

**Pull request merged**